src/public/modules/forms/admin/controllers/admin-form.client.controller.js

```diff
@@ -146,40 +185,27 @@ function AdminFormController(
         // and also updates myform if a formToUse is passed in
         $scope.myform = savedForm
       })
-      .catch((error) => {
-        if (!error) {
-          return
-        }
-        let errorMessage
-        switch (error.status) {
-          case HttpStatus.BAD_REQUEST:
-            errorMessage =
-              'This page seems outdated, and your changes could not be saved. Please refresh.'
-            break
-          case HttpStatus.UNAUTHORIZED:
-            errorMessage =
-              'Your changes could not be saved as your account lacks the requisite privileges.'
-            break
-          case HttpStatus.UNPROCESSABLE_ENTITY:
-            // Validation can fail for many reasons, so return more specific message
-            errorMessage = _.get(
-              error,
-              'data.message',
-              'Your changes contain invalid input.',
-            )
-            break
-          case HttpStatus.REQUEST_TOO_LONG: // HTTP Payload Too Large
-            errorMessage = `
-                Your form is too large. Reduce the number of fields, or submit a
-                <a href="${$scope.supportFormLink}" target="_blank" rel="noopener"><u>Support Form</u></a>.
-              `
-            break
-          default:
-            errorMessage = 'An error occurred while saving your changes.'
-        }
-        Toastr.error(errorMessage)
-        return error
+      .catch(handleUpdateError)
+  }
+
+  /**
+   * Calls the update form settings API
+   * @param {Object} settingsToUpdate the object with new values for settings.
+   * @returns Promise
+   */
+  $scope.updateFormSettings = (settingsToUpdate) => {
+    return $q
+      .when(
+        AdminFormService.updateFormSettings(
+          $scope.myform._id,
+          settingsToUpdate,
+        ),
+      )
+      .then((updatedSettings) => {
+        // merge back into main form since updating settings only returns the changed subset.
+        $scope.myform = merge($scope.myform, updatedSettings)
```

Comment:
Would this also work?
```suggestion
 Object.assign($scope.myform, updatedSettings)
```

Reply:
It should, since we are just returning the full subset of settings

Reply:
Done in 442e379